QHyper/optimizers/optimizer.py

```diff
@@ -0,0 +1,93 @@
+from abc import ABC, abstractmethod
+from typing import Callable, Any
+
+
+ArgsType = float | list[float]
+
+
+class Optimizer(ABC):
+    """Interface for optimizers"""
+
+    @abstractmethod
+    def minimize(
+        self, 
+        func: Callable[[ArgsType], float], 
+        init: ArgsType
+    ) -> ArgsType:
+        """Returns params which leads to the lowest value of the provided function 
+
+        Parameters
+        ----------
+        func : Callable[[ArgsType], float]
+            function, which will be minimize
+        init : ArgsType
+            initial args for optimizer
+
+        Returns
+        -------
+        ArgsType
+            Returns args which gave the lowest value.
+        """
+        pass
+
+
+class HyperparametersOptimizer(ABC):
+    """Interface for hyperoptimizers"""
+
+
+    def minimize(
+        self, 
+        func_creator: Callable[[ArgsType], Callable[[ArgsType], float]], 
+        optimizer: Optimizer, 
+        init: ArgsType, 
+        hyperparams_init: ArgsType, 
+        bounds: list[float],
+        **kwargs: Any
+    ) -> ArgsType:
+        """Returns hyperparameters which leads to the lowest values returned by optimizer 
+        
+        Parameters
+        ----------
+        func_creator : Callable[[ArgsType], Callable[[ArgsType], float]]
+            function, which receives hyperparameters, and returns 
+            function which will be optimized using optimizer
+        optimizer : Optimizer
+            object of class Optimizer
+        init : ArgsType
+            initial args for optimizer
+        hyperparams_init : ArgsType
+            initial hyperparameters
+        bounds : list[float]
+            bounds for hyperparameters
+        kwargs : Any
+            allow additional arguments
+
+        Returns
+        -------
+        ArgsType
+            hyperparameters which leads to the lowest values returned by optimizer
+        """
+        pass
+
+
+class Worker: #change name
```

Comment:
todo rename

Reply:
to what? xd I don't have any good idea :/